### DISCUSSION TITLE
Change wagmi connector's loginParams without recreating the connector?

### DISCUSSION THREAD

**Discussion by ryanberckmans** [Web3Auth/Web3Auth#1323] [Help]
Given a `Web3AuthConnector` from `"@web3auth/web3auth-wagmi-connector"` with this instantiation:
```typescript
const connector = new Web3AuthConnector({
  chains,
  options: {
    web3AuthInstance, // previously-constructed instance of Web3AuthCore
    loginParams: {
      loginProvider: 'google',
    },
  },
});
```
And assuming this connector has **already been added** to the singleton `wagmi.Client`

What's the recommended way to switch this connector instance to a new loginProvider, such as facebook or email_passwordless?

Does the Web3AuthConnector need to be recreated (reinstantiated) to change the loginProvider, or can the existing instance (that's already been added to wagmiClient and may be active and connected) have its login provider changed?

**Comment by brianonchain** [comment]
I am having the same issues. Were you able to find a solution? I'm thinking of just creating a connector for each of the social logins. Not sure if that's a good idea.
